### PR TITLE
[REEF-956] Install xUnit packages

### PR DESCRIPTION
**Pull request by tcNickolas**
This change:
 * Installs xUnit packages: unit testing framework and runners for console and VS
 * Adds missing license to several .csproj files

JIRA:
  [REEF-956](https://issues.apache.org/jira/browse/REEF-956)

Pull request:
  This closes #

### CODE REVIEW

**Comment by tcNickolas** [comment]
There's one more change Visual Studio does automatically whenever it tries to build this:

--- a/lang/cs/.nuget/NuGet.Config
+++ b/lang/cs/.nuget/NuGet.Config
@@ -21,11 +21,7 @@ under the License.
   <solution>
     <add key="disableSourceControlIntegration" value="true" />
   </solution>
-  <packageSources>
-      <add key="NuGet official package source" value="https://nuget.org/api/v2/" />
-  </packageSources>
-  <disabledPackageSources />
   <activePackageSource>
-      <add key="All" value="(Aggregate source)"  />
+      <add key="All" value="(Aggregate source)" />
   </activePackageSource>

Is this something we want to add as well?

File: lang/cs/Org.Apache.REEF.Client.Tests/Org.Apache.REEF.Client.Tests.csproj [hunk 1]
```diff
@@ -45,6 +47,22 @@ under the License.
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.ServiceProcess" />
+    <Reference Include="xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
+      <HintPath>..\packages\xunit.abstractions.2.0.0\lib\net35\xunit.abstractions.dll</HintPath>
```
Comment (anupam128) [inline lang/cs/Org.Apache.REEF.Client.Tests/Org.Apache.REEF.Client.Tests.csproj]: We use an alias like $(PackagesDir) for ..\packages.

**Comment by anupam128** [comment]
Do we need XUnit packages for non unittest projects?

File: lang/cs/.nuget/packages.config [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (anupam128) [inline lang/cs/.nuget/packages.config]: what is this file for?

**Comment by anupam128** [comment]
I have taken a look.

File: lang/cs/.nuget/packages.config (hunk 2 above)
Comment (tcNickolas) [inline lang/cs/.nuget/packages.config]: XUnit console runner, used to run tests from console, is solution-level package, so it adds itself here.

**Comment by tcNickolas** [comment]
We don't need xUnit packages for packages which don't have unittests, but I'm not sure that in the future we won't add tests to packages which don't currently have them... I guess we can limit current xUnit installations to the projects which currently have tests.
Org.Apache.REEF.Tang.Tests
Org.Apache.REEF.Wake.Tests
Org.Apache.REEF.Tests
Org.Apache.REEF.Common.Tests
Org.Apache.REEF.Network.Tests
Org.Apache.REEF.IMRU.Tests
Org.Apache.REEF.IO.Tests
Org.Apache.REEF.Evaluator.Tests
Org.Apache.REEF.Client.Tests

File: lang/cs/.nuget/NuGet.Config [hunk 3]
```diff
@@ -21,11 +21,7 @@ under the License.
   <solution>
     <add key="disableSourceControlIntegration" value="true" />
   </solution>
-  <packageSources>
```
Comment (jwang98052) [inline lang/cs/.nuget/NuGet.Config]: why remove?

**Comment by jwang98052** [comment]
Looks good in general. Any possibility to put those in a prop file at root folder then import in test's csproj file?

File: lang/cs/.nuget/NuGet.Config (hunk 3 above)
Comment (tcNickolas) [inline lang/cs/.nuget/NuGet.Config]: When I installed xUnit, it asked me to update to later nuget version. Now that I've updated it, nuget does this change automatically at every build. I don't think it makes a lot of sense to keep this file as it was.

**Comment by tcNickolas** [comment]
I'm not sure it makes a lot of sense to fight automatically generated project files. And then there might be issues with the order of imports. I'd rather keep it as is

**Comment by anupam128** [comment]
NuGet has good automation with UI and PowerShell. I agree with @tcNickolas to let automatically generated things as they are.

**Comment by tcNickolas** [comment]
I've resolved conflicts.

File: lang/cs/.nuget/packages.config (hunk 2 above)
Comment (jwang98052) [inline lang/cs/.nuget/packages.config]: Looks like those lines define default NuGet source. Not sure why the XUnit doesn't want it?

**Comment by jwang98052** [comment]
I am fine for leaving the references in individual test projects. 

Just for your info, in general, we don't always accept auto generated project code. We created build.prop file, the main reason is to unify the way in our projects instead simply accepting default. For example, VS auto generated script which contains something like All CPU and x86, it is very easy to cause run time issue. I believe that is one of the reasons most teams create their own base level prop files to avoid confusions.

**Comment by jwang98052** [comment]
Will test and merge

**Pull request closed without merging**